src/Aspire.Dashboard/Components/Pages/ComponentTelemetryContext.cs

```diff
@@ -15,9 +16,10 @@ public class ComponentTelemetryContext(string componentType) : IDisposable
 
     private DashboardTelemetryService TelemetryService => _telemetryService ?? throw new ArgumentNullException(nameof(_telemetryService), "InitializeAsync has not been called");
 
-    public async Task InitializeAsync(DashboardTelemetryService telemetryService)
+    public async Task InitializeAsync(DashboardTelemetryService telemetryService, IJSRuntime js)
```

Comment:
Inject IJSRuntime into the telemetry service so it doesn't need to be passed in each time? Or does DashboardTelemetryService have the wrong lifetime (singleton and not session?)